bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -196,20 +204,39 @@ public void run() {
 
                         if (lostBookies.size() > 0) {
                             knownBookies.removeAll(lostBookies);
-
-                            auditBookies();
+                            if (conf.getLostBookieRecoveryDelay() == 0 ||
+                                lostBookies.size() > 1 ||
+                                auditTask != null) {
+                                // 1) If more than one bookie is down, start the audit immediately;
+                                // 2) If we had scheduled an audit earlier for a lost bookie and in
+                                //    the meantime another bookie goes down, let us not delay recovery;
+                                //    we cancel the previously scheduled audit before starting the new one
+                                if (auditTask != null) {
+                                    if (auditTask.cancel(false)) {
```

Comment:
if the auditTask can not be cancelled, you will be starting another audit task, no?